opts/global.go

```diff
@@ -374,6 +376,10 @@ type AxisPointer struct {
 	Link []AxisPointerLink `json:"link,omitempty"`
 
 	Axis string `json:"axis,omitempty"`
+
+	Show bool `json:"show,omitempty"`
+
```

Comment:
TBH, I have a little struggling about the `bool` thing recently since the default behavior is totally opposite for the echats.i.e.
When we do not set `Show`
  In json result, it gonna be `false`, but the expectation is `true` or not exist in json as default `true` to echarts.

Maybe I need change all the `bool` config to `*bool` one day or something.
Currently the solution is we enrich the `json tag` with `default` and set those default `bool` if needs. 😅 

This new line `Show bool `json:"show,omitempty"` change hits me again. 😅

Reply:
I have dealt with this [deeply in another context](https://github.com/grafana/agent/issues/2607).  Go is awesome, but the default value handling is much trickier than in other languages (unless your default is `nil`!)
And I totally didn't notice it until you just pointed it out, since I always set it to true!

Pointer would be one way -- maybe some `DefaultTrueBool` type that the `json` marshaller handles and isn't tricky?   I do see this [default annotation](https://github.com/go-echarts/go-echarts/blob/master/opts/global.go#L869)... does that work?

Reply:
> I have dealt with this [deeply in another context](https://github.com/grafana/agent/issues/2607).  Go is awesome, but the default value handling is much trickier than in other languages (unless your default is `nil`!)
And I totally didn't notice it until you just pointed it out, since I always set it to true!

Atchoo, Same pain, same ppl. thats why I consider to refactor `bool` to `*bool` as well as those primitive types since it gonna be `nil` by default.

> Pointer would be one way -- maybe some `DefaultTrueBool` type that the `json` marshaller handles and isn't tricky?   I do see this [default annotation](https://github.com/go-echarts/go-echarts/blob/master/opts/global.go#L869)... does that work?  [edit to add: I mean does it work for `bool`, I see you have it for `string`.]

Yep, I consider to get a wrapper type for those primitive types such as `Bool`, `Integer`... but this gonna a breaking change as well.  and I m not quit sure whether "` DefaultTrueBool type that the  marshaller handles and isn't tricky?  ` " should work or not. I would appreciate that you could work with me on it. :)

About the `default` tag, it works when user does not set any opt of the opts. i.e.

If the do not set the `AxisPointer`, it will be set to true by `default true` tag. If user set one of the opts in `AxisPointer `, it won't work... so I think the `default tag` stuff is not very useful. hence I reconsider about something like the `DefaultTrueBool` or `Bool`.



> Also happy to remove this `Show` field from the PR.

I think it is reasonable to keep all the meaningful config same with echarts in go-echarts. so we should keep it and find a solution to fix the gap.

Reply:
I was reading this about [Optional JSON Fields in Go](https://eli.thegreenplace.net/2020/optional-json-fields-in-go/) and realized... since the Go default is opposite the JavaScrip default, we cannot mark this `bool` as `omit empty`.  Otherwise, when Golang specifies it as `false`, the `Marshal` won't emit the `show==false` then eCharts defaults it to true.  I will fix this.

Reply:
Ugh, as I implement this, I realize this `Show` can't be added or it will be a breaking change as well... they will need to set it or their stuff will disappear.   so the "default style" per [#287](https://github.com/go-echarts/go-echarts/issues/287)should be figured out first.

Reply:
I just opted to go the pointer path.. to try it on...  and happy so far.

Reply:
> I just opted to go the pointer path.. to try it on... and happy so far.

yea, the #287 should be done asap. and I think in this PR put the `bool` now is fine since we gonna refactor all the primitive types later when the #287 is done.